### ISSUE TITLE
Inconsistent behavior of ImportOrderCheck

### ISSUE THREAD

**Issue by ivanov-alex** [checkstyle/checkstyle#1398]
In some cases ImportOrderCheck requires line separator between static imports, in some cases not.
Checkstyle 6.8 Config:
```xml
<module name="ImportOrder">
    <property name="groups" value="/^javax?\./,org"/>
    <property name="ordered" value="true"/>
    <property name="separated" value="true"/>
    <property name="option" value="top"/>
    <property name="sortStaticImportsAlphabetically" value="true"/>
</module>
```

```java
import static com.puppycrawl.tools.checkstyle.ImportOrderCheck.MSG_SEPARATION;
import static javax.lang.model.element.ElementKind.CONSTRUCTOR;
import static org.androidannotations.helper.ModelConstants.generationSuffix;
import static net.sourceforge.pmd.lang.ast.Node; 
// /test.java:5: warning: 'org.androidannotations.helper.ModelConstants.generationSuffix' should be separated from previous imports.
// /test.java:6: warning: 'net.sourceforge.pmd.lang.ast.Node' should be separated from previous imports.

import java.util.Set;

import org.*;

import com.puppycrawl.tools.checkstyle.ImportOrderCheck;
```
```java
import static com.puppycrawl.tools.checkstyle.ImportOrderCheck.MSG_SEPARATION;
import static javax.lang.model.element.ElementKind.CONSTRUCTOR;
import static net.sourceforge.pmd.lang.ast.Node; 
// /test.java:6: warning: 'net.sourceforge.pmd.lang.ast.Node' should be separated from previous imports.

import java.util.Set;

import org.*;

import com.puppycrawl.tools.checkstyle.ImportOrderCheck;
```
```java
import static com.puppycrawl.tools.checkstyle.ImportOrderCheck.MSG_SEPARATION;
import static javax.lang.model.element.ElementKind.CONSTRUCTOR;
// no violations
import java.util.Set;

import org.*;

import com.puppycrawl.tools.checkstyle.ImportOrderCheck;
```
```java
import static com.puppycrawl.tools.checkstyle.ImportOrderCheck.MSG_SEPARATION;
import static org.androidannotations.helper.ModelConstants.generationSuffix;
// no violations
import java.util.Set;

import org.*;

import com.puppycrawl.tools.checkstyle.ImportOrderCheck;
```
```java
import static com.puppycrawl.tools.checkstyle.ImportOrderCheck.MSG_SEPARATION;
import static net.sourceforge.pmd.lang.ast.Node;
// no violations
import java.util.Set;

import org.*;

import com.puppycrawl.tools.checkstyle.ImportOrderCheck;
```
```java
import static javax.lang.model.element.ElementKind.CONSTRUCTOR;
import static net.sourceforge.pmd.lang.ast.Node; 
// /test.java:6: warning: 'net.sourceforge.pmd.lang.ast.Node' should be separated from previous imports.

import java.util.Set;

import org.*;

import com.puppycrawl.tools.checkstyle.ImportOrderCheck;
```

**Comment by ivanov-alex** [comment]
Based on [this code](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java#L347) I don't think that there was intention to raise separation violations within "import static" group (because that there is a special method to process imports within a group, it is issued when import is from the same group or when we have static imports to be sorted on top or bottom).

**Comment by ivanov-alex** [comment]
Short issue description: CS requires separator after first static import from the group defined on config. But it doesn't require separator before import this import.
Lets assume we have only java.* group.

Now we DON'T HAVE violation in:
```java
import static com.puppycrawl.tools.checkstyle.ImportOrderCheck.MSG_SEPARATION;
import static java.lang.model.element.ElementKind.CONSTRUCTOR;
```
And HAVE violation in
```java
import static java.lang.model.element.ElementKind.CONSTRUCTOR;
import static com.puppycrawl.tools.checkstyle.ImportOrderCheck.MSG_SEPARATION;
```

Correct behavior is to have identical behavior in both cases.

**Comment by romani** [comment]
@ivanov-alex , can you explain why we specify in configuration "javax,org" but order of imports is : 
```
import org.*;
import java.util.Set;
import com.puppycrawl.tools.checkstyle.ImportOrderCheck;
```

**Comment by ivanov-alex** [comment]
Corrected initial comment to have non-static imports sorted correctly.
While preparing original messages, I was paying attention to static imports only and skipped violations for non-static. Now there should be no logical conflicts.

**Comment by romani** [comment]
merged.
